pkg/cmd/cli/cmd/set/triggers.go

```diff
@@ -0,0 +1,744 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+const (
+	triggersLong = `
+Set or remove triggers for build configs and deployment configs
+
+All build configs and deployment configs may have a set of triggers that result in a new deployment
+or build being created. This command enables you to alter those triggers - making them automatic or
+manual, adding new entries, or changing existing entries.
```

Comment:
I'd rather remove --remove and have people use --manual.  And then we open
the bug on the build config that we can't express what we actually want.

On Wed, Feb 24, 2016 at 1:19 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/set/triggers.go
> <https://github.com/openshift/origin/pull/7423#discussion_r53981638>:
>
> > +	buildutil "github.com/openshift/origin/pkg/build/util"
> > +	cmdutil "github.com/openshift/origin/pkg/cmd/util"
> > +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +	deployapi "github.com/openshift/origin/pkg/deploy/api"
> > +	"github.com/openshift/origin/pkg/generate/app"
> > +	imageapi "github.com/openshift/origin/pkg/image/api"
> > +	"k8s.io/kubernetes/pkg/util/sets"
> > +)
> > +
> > +const (
> > +	triggersLong = `
> > +Set or remove triggers for build configs and deployment configs
> > +
> > +All build configs and deployment configs may have a set of triggers that result in a new deployment
> > +or build being created. This command enables you to alter those triggers - making them automatic or
> > +manual, adding new entries, or changing existing entries.
>
> they are different, users should know to treat them differently. "where
> did my trigger go? all i did was put it in manual mode and now it's gone!"
> doesn't seem like what i'd want.
>
> "oh, buildconfig triggers don't support manual mode so i have to remove
> them...that sucks, but now i know" does.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7423/files#r53981638>.
>